main.tf

```
@@ -0,0 +1,144 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+}
+
+// Region specified in AWS provider
+data "aws_region" "current" {}
+
+resource "aws_secretsmanager_secret" "prefect_api_key" {
+  name                    = "prefect-api-key-${var.name}"
+  recovery_window_in_days = var.secrets_manager_recovery_in_days
+}
+
+resource "aws_secretsmanager_secret_version" "prefect_api_key_version" {
+  secret_id     = aws_secretsmanager_secret.prefect_api_key.id
+  secret_string = var.prefect_api_key
+}
+
+resource "aws_iam_role" "prefect_worker_execution_role" {
+  name = "prefect-worker-execution-role-${var.name}"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "prefect_worker_ecs_policy" {
+  role = aws_iam_role.prefect_worker_execution_role.name
+
+  // AmazonECSTaskExecutionRolePolicy is an AWS managed role for creating ECS tasks and services
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role_policy" "ssm_allow_read_prefect_api_key" {
+  name = "ssm-allow-read-prefect-api-key-${var.name}"
+  role = aws_iam_role.prefect_worker_execution_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "kms:Decrypt",
+          "secretsmanager:GetSecretValue",
+          "ssm:GetParameters"
+        ]
+        Effect = "Allow"
+        Resource = [
+          aws_secretsmanager_secret.prefect_api_key.arn
+        ]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "logs_allow_create_log_group" {
```

```suggestion
resource "aws_iam_role_policy" "allow_create_log_group" {
```

the type is the iam policy, the name can be what the policy allows you to do

---

Nice, thanks - updated in 6915a93831e